windows/ReactTestApp/ReactTestApp.vcxproj

```diff
@@ -116,104 +141,113 @@
+    <ItemGroup>
+        <Text Include="$(ManifestRootPath)\app.json" />
     </ItemGroup>
     <ItemGroup>
-        <ClCompile Include="pch.cpp">
+        <ClCompile Include="$(SourceFilesPath)\pch.cpp">
             <PrecompiledHeader>Create</PrecompiledHeader>
         </ClCompile>
-        <ClCompile Include="App.cpp">
-            <DependentUpon>App.xaml</DependentUpon>
+        <ClCompile Include="$(SourceFilesPath)\App.cpp">
+            <DependentUpon>$(SourceFilesPath)\App.xaml</DependentUpon>
         </ClCompile>
-        <ClCompile Include="MainPage.cpp">
-            <DependentUpon>MainPage.xaml</DependentUpon>
+        <ClCompile Include="$(SourceFilesPath)\MainPage.cpp">
+            <DependentUpon>$(SourceFilesPath)\MainPage.xaml</DependentUpon>
         </ClCompile>
         <ClCompile Include="$(GeneratedFilesDir)module.g.cpp" />
-        <ClCompile Include="Manifest.cpp" />
-        <ClCompile Include="ReactInstance.cpp" />
+        <ClCompile Include="$(SourceFilesPath)\Manifest.cpp" />
+        <ClCompile Include="$(SourceFilesPath)\ReactInstance.cpp" />
     </ItemGroup>
     <ItemGroup>
-        <Midl Include="App.idl">
-            <DependentUpon>App.xaml</DependentUpon>
+        <Midl Include="$(SourceFilesPath)\App.idl">
+            <DependentUpon>$(SourceFilesPath)\App.xaml</DependentUpon>
         </Midl>
-        <Midl Include="MainPage.idl">
-            <DependentUpon>MainPage.xaml</DependentUpon>
+        <Midl Include="$(SourceFilesPath)\MainPage.idl">
+            <DependentUpon>$(SourceFilesPath)\MainPage.xaml</DependentUpon>
         </Midl>
     </ItemGroup>
     <ItemGroup>
         <None Include="packages.config" />
         <None Include="PropertySheet.props" />
     </ItemGroup>
     <ItemGroup>
-        <BundleContentToBePackaged Include="$(BundleContent)" />
-        <None Include="@(BundleContentToBePackaged)">
+        <BundleDirContentToBePackaged Include="$(BundleDirAssets)"/>
+        <None Include="@(BundleDirContentToBePackaged)">
+            <DirNameFullPath>$([MSBuild]::ValueOrDefault('%(FullPath)', '').Remove($([MSBuild]::ValueOrDefault(%(FullPath), '').LastIndexOf(\%(RecursiveDir)%(Filename)%(Extension)))))</DirNameFullPath>
+            <DirName Condition="'%(DirNameFullPath)' != ''">$([MSBuild]::ValueOrDefault(%(DirNameFullPath), '').Substring($([MSBuild]::ValueOrDefault(%(DirNameFullPath), '').LastIndexOf('\'))))</DirName>
+            <Link>$([MSBuild]::MakeRelative($(ProjectDir),'Bundle%(DirName)\%(RecursiveDir)%(Filename)%(Extension)'))</Link>
+            <DeploymentContent>true</DeploymentContent>
+        </None>
+    </ItemGroup>
+    <ItemGroup>
+        <BundleFileContentToBePackaged Include="$(BundleFileAssets)"/>
+        <None Include="@(BundleFileContentToBePackaged)">
             <Link>$([MSBuild]::MakeRelative($(ProjectDir),'Bundle\%(RecursiveDir)%(Filename)%(Extension)'))</Link>
             <DeploymentContent>true</DeploymentContent>
         </None>
     </ItemGroup>
```

Comment:
This can probably be written in a less cumbersome way. What we try to achieve here is the following:

- When resource entry in manifest is a folder, we want to include the folder itself, not only its contents. For example, for entry `dist/assets` we want the paths to included files to be `Bundle/assets/...`, not just `Bundle/...`
- When entry i a file, we want to include just the file itself (without putting it in an additional directory). So for entry `dist/main.bundle`, the path to included file should be `Bundle/main.bundle`, not `Bundle/dist/main.bundle`

However, I'm not sure how to generalize this.

Reply:
Normally, we just want to reference the last "part" of the path. E.g. for `dist/assets`, what we really want to be part of the bundle is `assets`. For a longer path, say `dist/resources/myAsset.res`, we should only reference `myAsset.res`. The final bundle would look something like:
```
Bundle
├── assets
│   └── <content of assets>
├── main.windows.bundle
└── myAsset.res
```
We're keeping only the folder structure inside the referenced folder. Otherwise, we will always flatten everything.

Reply:
That's what happens now as well. The problem is that to include all contents of `assets` we need to have `assets\**\*` as include path, which would then match each path inside `assets`. So the last "part" will actually be filename.